libkbfs/journal_server.go

```diff
@@ -45,8 +45,9 @@ type JournalServer struct {
 	delegateBlockServer BlockServer
 	delegateMDOps       MDOps
 
-	lock        sync.RWMutex
-	tlfJournals map[TlfID]*tlfJournal
+	lock            sync.RWMutex
```

Comment:
you could avoid a bunch of complexity communicating branch changes, I think, by doing the following:

1) Define an interface like

```
type branchChangeDelegate interface {
  onTLFBranchChange(TlfID, BranchID)
}
```

2) Make `KBFSOps` implement this interface, and have it route to the appropriate FBO.

3) Have `JournalServer` take a `branchChangeDelegate` and pass it unchanged to its `tlfJournal`s. Have `tlfJournal` call it as needed (either spawn a goroutine, or mandate that implmentations shouldn't block, etc.). Then pass in the `KBFSOps` to `makeJournalServer` (or use `config.KBFSOps()`, if it implements the interface).

What do you think?